src/components/trade/index.tsx

```diff
@@ -169,148 +180,163 @@ export const TradeEntry = () => {
   );
 };
 
-
-export const TradeInfo = (props: {
-  pool?: PoolInfo;
-}) => {
-  const {
-    A,
-    B,
-    lastTypedAccount,
-  } = useCurrencyPairState();
+export const TradeInfo = (props: { pool?: PoolInfo }) => {
+  const { A, B } = useCurrencyPairState();
   const { pool } = props;
   const { slippage } = useSlippageConfig();
   const pools = useMemo(() => (pool ? [pool] : []), [pool]);
   const enriched = useEnrichedPools(pools);
 
   const [amountOut, setAmountOut] = useState(0);
-  const [maxMinLabel, setMaxMinLabelOut] = useState("");
   const [priceImpact, setPriceImpact] = useState(0);
   const [lpFee, setLpFee] = useState(0);
   const [exchangeRate, setExchangeRate] = useState(0);
   const [priceAccount, setPriceAccount] = useState("");
 
   useEffect(() => {
-
     if (!pool || enriched.length === 0) {
       return;
     }
-    if(B.amount) {
+    if (B.amount) {
       const minAmountOut = parseFloat(B?.amount) * (1 - slippage);
       setAmountOut(minAmountOut);
-      setMaxMinLabelOut("Minimun Received");
     }
-    // else if(A.amount && lastTypedAccount === B.mintAddress) {
-    //   const maxAmountOut = ;
-    //   setAmountOut(maxAmountOut);
-    //   setMaxMinLabelOut("Maximun Sold");
-    // }
     const liqA = enriched[0].liquidityA;
     const liqB = enriched[0].liquidityB;
-    const supplyRatio = (liqA / liqB);
+    const supplyRatio = liqA / liqB;
     // We need to make sure the order matched the pool's accounts order
     const enrichedA = A.mintAddress === enriched[0].mints[0] ? A : B;
     const enrichedB = enrichedA.mintAddress === A.mintAddress ? B : A;
-    const calculatedRatio = (
-      parseFloat(enrichedA.amount) / parseFloat(enrichedB.amount)
-    );
+    const calculatedRatio =
+      parseFloat(enrichedA.amount) / parseFloat(enrichedB.amount);
     // % difference between pool ratio and  calculated ratio
-    setPriceImpact(Math.abs(100 - (calculatedRatio*100) / supplyRatio))
+    setPriceImpact(Math.abs(100 - (calculatedRatio * 100) / supplyRatio));
 
     // 6 decimals without trailing zeros
-    const lpFeeStr = (parseFloat(A.amount) * LIQUIDITY_PROVIDER_FEE).toFixed(6)
-    setLpFee(parseFloat(lpFeeStr))
-  }, [A, B, slippage, lastTypedAccount, pool, enriched]);
+    const lpFeeStr = (parseFloat(A.amount) * LIQUIDITY_PROVIDER_FEE).toFixed(6);
+    setLpFee(parseFloat(lpFeeStr));
 
-  useEffect(() => {
-    if(priceAccount === B.mintAddress) {
-      setExchangeRate(parseFloat(B.amount) / parseFloat(A.amount))
+    if (priceAccount === B.mintAddress) {
+      setExchangeRate(parseFloat(B.amount) / parseFloat(A.amount));
     } else {
-      setExchangeRate(parseFloat(A.amount) / parseFloat(B.amount))
+      setExchangeRate(parseFloat(A.amount) / parseFloat(B.amount));
     }
-  }, [A, B, priceAccount])
+  }, [B, slippage, pool, enriched, priceAccount]);
```

Comment:
why is A not in the deps list?

Reply:
There was a bug in the text. All the numbers flashed with weird values. So the problem is when you change A, B is changed Async. So if you had any value on B, the rate A / B wasn't updated right away and the numbers changed till B is updated.

Reply:
I'm gonna Add `A` again to the deps, and work on the issue